### DISCUSSION TITLE
Owner ID = 0 in Standard Information

### DISCUSSION THREAD

**Discussion by whitequill** [ColinFinck/ntfs#29] [Q&A]
Here is my example:
```
PS C:\Windows\System32> get-acl ..\CSC\v2.0.6\

    Directory: C:\Windows\CSC

Path   Owner               Access
----   -----               ------
v2.0.6 NT AUTHORITY\SYSTEM NT AUTHORITY\SYSTEM Allow  FullControl

PS C:\Windows\System32> Get-Acl C:\Users\brazg\Documents\JoyLang\

    Directory: C:\Users\brazg\Documents

Path    Owner     Access
----    -----     ------
JoyLang MSI\brazg NT AUTHORITY\SYSTEM Allow  FullControl…
```

But when I run the ntfs-shell I get:
```
ntfs-shell:\Windows> fileinfo CSC
============================= FILE RECORD ==============================
Allocated Size:                   1024
Byte Position:                    0xc841c000
Data Size:                        672
Hard-Link Count:                  1
Is Directory:                     true
Record Number:                    0x21070
Sequence Number:                  21

========================= STANDARD INFORMATION =========================
Attributes:
Access Time:                      2023-07-08 10:14:39 UTC
Creation Time:                    2023-06-28 06:20:07 UTC
Modification Time:                2023-06-28 06:22:02 UTC
MFT Record Modification Time:     2023-07-08 08:20:12 UTC
Class ID:                         0
Maximum Versions:                 0
Owner ID:                         0
Quota Charged:                    0
Security ID:                      267
USN:                              5465278264
Version:                          0

============================== FILE NAME ===============================
Name:                             "CSC"
Namespace:                        Win32AndDos
Parent Directory Record Number:   0x894b0
```

```
ntfs-shell:\Users\brazg\Documents\JoyLang> fileinfo joy.tar.gz
============================= FILE RECORD ==============================
Allocated Size:                   1024
Byte Position:                    0xc65ca000
Data Size:                        456
Hard-Link Count:                  2
Is Directory:                     false
Record Number:                    0x19728
Sequence Number:                  23

========================= STANDARD INFORMATION =========================
Attributes:                       ARCHIVE
Access Time:                      2023-07-07 13:00:10 UTC
Creation Time:                    2023-03-10 12:02:54 UTC
Modification Time:                2022-03-25 10:24:44 UTC
MFT Record Modification Time:     2023-03-10 12:02:54 UTC
Class ID:                         0
Maximum Versions:                 0
Owner ID:                         0
Quota Charged:                    0
Security ID:                      2445
USN:                              2693536488
Version:                          0

============================== FILE NAME ===============================
Name:                             "joy.tar.gz"
Namespace:                        Win32
Parent Directory Record Number:   0x1971e

============================== FILE NAME ===============================
Name:                             "JOYTAR~1.GZ"
Namespace:                        Dos
Parent Directory Record Number:   0x1971e

============================= DATA STREAM ==============================
Name:                             ""
Size:                             75647
```

Why do thay both show a OwnerID of zero? Is this the expected behavior?

Also on a tangent is there a way to run the `fileinfo` command on the root directory of C its self?

**Comment by ColinFinck** [comment]
> Why do thay both show a OwnerID of zero? Is this the expected behavior?

Yes. As per https://flatcap.github.io/linux-ntfs/ntfs/attributes/standard_information.html, the "Owner Id" of the `$STANDARD_INFORMATION` attribute is only relevant for the Quota feature that Windows 2000 introduced. If "Owner Id" is zero, the Quota feature is disabled.

What you are looking for are the way more complex ACL permissions of a file. They are partly referenced by the "Security Id" attribute that `fileinfo` outputs, but you also need additional information that is stored in the root `$Secure` file.

I haven't got around yet to implement comfortable support for reading ACLs in `ntfs`.
If you wanted to do that, you would currently need to read the top-level `$Secure` file and its streams manually and find the ACL that corresponds to your "Security Id". https://www.ntfs.com/ntfs-permissions-file-structure.htm is a great starting point on that topic.

> Also on a tangent is there a way to run the `fileinfo` command on the root directory of C its self?

Either `fileinfo .` (in the root) or `fileinfo /5` (works anywhere). The latter syntax references a file by its global File Record Number (see `help fileinfo`). The File Record Number of the root directory is always 5.

**Comment by whitequill** [comment]
Firstly, thank you for your response. I had seen the project was active but I had also noticed there were very few discussions or questions being asked about the project.
I came to your project thought this News Letter from ReactOS: https://reactos.org/blogs/newsletter-102/. I have been following ReactOS on and off for a number of years mostly though downloads and not their news letter.  I have also had a personal curiosity in ntfs and particularly Alternate Data Streams.  I've noticed that they don't seem to be used for anything hardly except for just showing if a file was downloaded from the internet or not and I'm curious why? 
I can't see why one wouldn't use put a DLL and EXE together into a single file with multiple data streams that is runnable. or place configuration information into an ADS as data-streams aren't limited like the limited extended attributes on Linux.

I have also run into a short coming that a file isn't reable unless its in the same directory as another one on windows so you need to place a link to the data-stream in the directory or its not _visible_ to other files.

Lastly, why do you need to use UNC path format when specifying a path?
